exec/java-exec/src/test/java/org/apache/drill/TestTpchDistributedConcurrent.java

```diff
@@ -0,0 +1,177 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill;
+
+import java.io.IOException;
+import java.util.IdentityHashMap;
+import java.util.Random;
+import java.util.concurrent.Semaphore;
+
+import org.apache.drill.QueryTestUtil;
+import org.apache.drill.common.exceptions.UserException;
+import org.apache.drill.common.util.TestTools;
+import org.apache.drill.exec.proto.UserBitShared;
+import org.apache.drill.exec.proto.UserBitShared.QueryResult.QueryState;
+import org.apache.drill.exec.rpc.user.UserResultsListener;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+/*
+ * Note that the real interest here is that the drillbit doesn't become
+ * unstable from running a lot of queries concurrently -- it's not about
+ * any particular order of execution. We ignore the results.
+ */
+// TODO(cwestin) @Ignore
+public class TestTpchDistributedConcurrent extends BaseTestQuery {
+  @Rule public final TestRule TIMEOUT = TestTools.getTimeoutRule(120000); // Longer timeout than usual.
+
+  /*
+   * Valid test names taken from TestTpchDistributed. Fuller path prefixes are
+   * used so that tests may also be taken from other locations -- more variety
+   * is better as far as this test goes.
+   */
+  private final static String queryFile[] = {
+    "queries/tpch/01.sql",
+    "queries/tpch/03.sql",
+    "queries/tpch/04.sql",
+    "queries/tpch/05.sql",
+    "queries/tpch/06.sql",
+    "queries/tpch/07.sql",
+    "queries/tpch/08.sql",
+    "queries/tpch/09.sql",
+    "queries/tpch/10.sql",
+    "queries/tpch/11.sql",
+    "queries/tpch/12.sql",
+    "queries/tpch/13.sql",
+    "queries/tpch/14.sql",
+    // "queries/tpch/15.sql", this creates a view
+    "queries/tpch/16.sql",
+    "queries/tpch/18.sql",
+    "queries/tpch/19_1.sql",
+    "queries/tpch/20.sql",
+  };
+
+  private final static int TOTAL_QUERIES = 115;
+  private final static int CONCURRENT_QUERIES = 15;
+
+  private final static Random random = new Random(0xdeadbeef);
+  private final static String alterSession = "alter session set `planner.slice_target` = 10";
+
+  private int remainingQueries = TOTAL_QUERIES - CONCURRENT_QUERIES;
+  private final Semaphore completionSemaphore = new Semaphore(0);
+  private final Semaphore submissionSemaphore = new Semaphore(0);
+  private final IdentityHashMap<UserResultsListener, Object> listeners = new IdentityHashMap<>();
```

Comment:
Is there any particular advantage to using that one instead? It sounds exactly the same.